### PR TITLE
PLANNER-2629 Pause best solution consumer

### PR DESCRIPTION
**Pull request by rsynek**
### JIRA
https://issues.redhat.com/browse/PLANNER-2629
https://issues.redhat.com/browse/PLANNER-2650

### Referenced pull requests
https://github.com/kiegroup/optaplanner/pull/1832

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add the label `run_fdb`
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] tests</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] native</b>
</details>

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - optaplanner-quickstarts job** `#103` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner-quickstarts.tests.optaplanner-quickstarts/103/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_BoxShadow.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_m_dark_BackgroundColor.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_m_expanded_Transform.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_md_Transform.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_md_Width.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_Transform.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_Transition.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_Width.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page__sidebar_ZIndex.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1644236326493/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/js/c_page_BackgroundColor.js’: No such file or directory<br/>[2022-02-07T12:18:51.184Z] 638M	/home/jenkins/workspace/KIE/kogito/main<br/>[2022-02-07T12:18:51.184Z] 638M	/home/jenkins/workspace/KIE/kogito<br/>[2022-02-07T12:18:51.184Z] 638M	/home/jenkins/workspace/KIE<br/>[2022-02-07T12:18:51.184Z] 638M	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-07T12:18:52.003Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner-quickstarts.tests.optaplanner-quickstarts/103/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java [hunk 1]
```diff
@@ -50,9 +52,20 @@ public TimeTable findById(Long id) {
     }
 
     public void save(TimeTable timeTable) {
+        try {
+            Thread.sleep(1000L);
```
Comment (rsynek) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java:56]: TODO: remove (helps to show that the `pauseConsumption()` actually works as expected)

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java [hunk 2]
```diff
@@ -50,9 +52,20 @@ public TimeTable findById(Long id) {
...
+        }
         for (Lesson lesson : timeTable.getLessonList()) {
             // TODO this is awfully naive: optimistic locking causes issues if called by the SolverManager
             lessonRepository.save(lesson);
+
+//            Optional<Lesson> attachedLesson = lessonRepository.findById(lesson.getId());
+//            if (attachedLesson.isPresent()) {
+//                attachedLesson.get().setTimeslot(lesson.getTimeslot());
+//                attachedLesson.get().setRoom(lesson.getRoom());
+//            }
```
Comment (rsynek) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java:68]: Interestingly, if we change the persistence the way it's done in the Quarkus quickstart, the race condition is gone, too.

**Review by triceo — APPROVED** [review]

**Comment by ge0ffrey** [comment]
We can't merge this PR, as it complexitfies our getting started experience (the gained features don't make up for the complexity).
But htis PR wasn't meant to be merged: it was meant to prove these functions work well.

**Pull request closed without merging**